### DISCUSSION TITLE
Is there any convenient way to convert milkdown's content into pdf?

### DISCUSSION THREAD

**Discussion by 9vek** [Milkdown/milkdown#428] [Q&A]
Hello guys,
Is there a convenient way to convert a milkdown instance's markdown content into pdf or doc file and keep all the styles?
I know this kind of operations may not suitable to be done at the client-side, all ways I found myself seems a bit complicated to me, are there any instructions or recommendations?

**Comment by Saul-Mirone** [comment]
The real problem is how to convert dom tree to pdf. Because you can get a readable dom tree from readonly mode. Maybe what you need is https://github.com/parallax/jsPDF

**Reply by 9vek** [reply]
That means to do this at the cilent-side? Actually I have tried using jsPDF and I realized that I have to write a dozen of code if I want to keep the styles exactly the same with the editor. Do you mean using jsPDF is probably the best practice at present, for milkdown-pdf convertion?
Thanks for your reply. :)

**Reply by Saul-Mirone** [reply]
I'm not sure if jsPDF is the best practice because I haven't investigate deeply in this feature. My point is any dom to pdf tools would be helpful.

**Reply by langsz** [reply]
Please focus on [wkhtmltopdf](https://wkhtmltopdf.org/)
